sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetScan.scala

```diff
@@ -70,22 +71,37 @@ abstract class GpuParquetScanBase(
     readPartitionSchema: StructType,
     pushedFilters: Array[Filter],
     rapidsConf: RapidsConf)
-  extends ScanWithMetrics {
+  extends ScanWithMetrics with Logging {
 
   def isSplitableBase(path: Path): Boolean = true
 
   def createReaderFactoryBase(): PartitionReaderFactory = {
     val broadcastedConf = sparkSession.sparkContext.broadcast(
       new SerializableConfiguration(hadoopConf))
-    GpuParquetPartitionReaderFactory(sparkSession.sessionState.conf, broadcastedConf,
-      dataSchema, readDataSchema, readPartitionSchema, pushedFilters, rapidsConf, metrics)
+    // We have to look at the latest conf because we set the input file exec used
+    // conf after the rapidsConf is passed in here.
+    val latestConf = new RapidsConf(sparkSession.sessionState.conf)
```

Comment:
We might be loading multiple files in a query and input file exec might only be used on one of the files.  Therefore we don't want to disable all small file processing, just on the one scan exec that needs it disabled.  Storing it in the SQLConf doesn't allow that granularity.  IIRC this is already done properly for GPUCoalesceBatches, seems like we could do similar processing in the transition overrides to "fixup" any scans to disable small file processing (as an exec construction parameter rather than a conf).

Reply:
thanks, I didn't see the code in GPUCoalesceBatches that did similar things, I'll move there